chapter4/src/main/java/com/backstopmedia/kotlin/chapter4/ui/NavigationHelper.kt

```diff
@@ -0,0 +1,172 @@
+package com.backstopmedia.kotlin.chapter4.ui
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.annotation.IdRes
+import android.support.annotation.LayoutRes
+import android.support.design.widget.NavigationView
+import android.support.v4.app.Fragment
+import android.support.v4.widget.DrawerLayout
+import android.support.v7.app.ActionBarDrawerToggle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.view.LayoutInflater
+import android.view.MenuItem
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import com.backstopmedia.kotlin.chapter4.R
+import com.backstopmedia.kotlin.chapter4.entities.Profile
+import com.backstopmedia.kotlin.chapter4.interactors.UserInteractor
+import com.backstopmedia.kotlin.chapter4.interactors.UserInteractorImpl
+import com.backstopmedia.kotlin.chapter4.presenters.NavigationDrawerPresenter
+import com.backstopmedia.kotlin.chapter4.presenters.NavigationDrawerPresenterImpl
+import com.backstopmedia.kotlin.chapter4.ui.activities.TimelineActivity
+import com.backstopmedia.kotlin.chapter4.ui.activities.TopImagesActivity
+import com.backstopmedia.kotlin.chapter4.ui.view.NavigationDrawerView
+import com.backstopmedia.kotlin.chapter4.utils.glide.CircleTransformation
+import com.bumptech.glide.Glide
+import com.twitter.sdk.android.core.TwitterCore
+import org.jetbrains.anko.find
+import kotlin.properties.Delegates
+
+/**
+ * Created by Tudor Luca on 13/10/15.
+ */
+class NavigationHelper : Fragment(), NavigationDrawerView {
+
+    companion object {
+        fun setup(activity: AppCompatActivity, @LayoutRes resId: Int): NavigationHelper {
+            val navigationHelper = activity.attachNavigationHelper()
+            navigationHelper.setContentViewForActivity(activity, resId)
+            return navigationHelper
+        }
+    }
+
+    private val interactor: UserInteractor by lazy { UserInteractorImpl(TwitterCore.getInstance().sessionManager.activeSession) }
+    private val presenter: NavigationDrawerPresenter  by lazy { NavigationDrawerPresenterImpl(interactor) }
+
+    private var isSubdecorInstalled = false
+    private var drawerLayout: DrawerLayout by Delegates.notNull()
```

Comment:
In this case do we want Delegates.notNull() or would it be better to use the lazyInit keyword?